src/DistributedLock.Etcd/EtcdClientWrapper.cs

```diff
@@ -0,0 +1,58 @@
+using dotnet_etcd;
+using dotnet_etcd.interfaces;
+using Etcdserverpb;
+using Grpc.Core;
+using Medallion.Threading.Internal;
+using V3Lockpb;
+
+namespace Medallion.Threading.Etcd;
+
+public class EtcdClientWrapper 
+{
+    private readonly IEtcdClient _etcdClient;
+    public EtcdClientWrapper(IEtcdClient etcdClient)
+    {
+        _etcdClient = etcdClient ?? throw new ArgumentNullException(nameof(etcdClient));
+    }
+
+    public ValueTask<LeaseGrantResponse> LeaseGrantAsync(LeaseGrantRequest request, CancellationToken cancellationToken)
+    {
+        return SyncViaAsync.IsSynchronous
+            ? new ValueTask<LeaseGrantResponse>(this._etcdClient.LeaseGrant(request,
+                cancellationToken: cancellationToken))
+            : new ValueTask<LeaseGrantResponse>(
+                this._etcdClient.LeaseGrantAsync(request, cancellationToken: cancellationToken));
+    }
+
+    public Task LeaseKeepAlive(long leaseId, CancellationToken token)
+    {
+        return this._etcdClient.LeaseKeepAlive(leaseId, token);
+    }
+
+    public async ValueTask<LockResponse> LockAsync(LockRequest lockRequest, CancellationToken cancellationToken)
+    {
+        // thread static var here is legit? TODO why not asynclocal??
```

Comment:
Yes it works because in an IsSynchronous flow you never leave the current thread. Therefore you don't need to flow the value across threads.

Let's remove this comment